go/test/endtoend/backup/vtctlbackup/backup_test.go

```diff
@@ -70,3 +81,132 @@ func setDefaultCompressionFlag() {
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", "--backup-engine=builtin", primary.Alias)
+	require.NoError(t, err)
+	engineUsed := getBackupEngineOfLastBackup(t)
+	require.Equal(t, "builtin", engineUsed)
+
+	// then try to backup specifying the xtrabackup engine
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", "--backup-engine=xtrabackup", primary.Alias)
+	require.NoError(t, err)
+	engineUsed = getBackupEngineOfLastBackup(t)
+	require.Equal(t, "xtrabackup", engineUsed)
+
+	// check that by default we still use the xtrabackup engine if not specified
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", primary.Alias)
+	require.NoError(t, err)
+	engineUsed = getBackupEngineOfLastBackup(t)
+	require.Equal(t, "xtrabackup", engineUsed)
+}
+
+// fetch the backup engine used on the last backup triggered by the end-to-end tests.
+func getBackupEngineOfLastBackup(t *testing.T) string {
+	lastBackup := getLastBackup(t)
+
+	// open the Manifest and retrieve the backup engine that was used
+	f, err := os.Open(path.Join(localCluster.CurrentVTDATAROOT,
+		"backups", keyspaceName, shardName,
+		lastBackup, "MANIFEST",
+	))
+	require.NoError(t, err)
+	defer f.Close()
+
+	data, err := io.ReadAll(f)
+	require.NoError(t, err)
+
+	var manifest mysqlctl.BackupManifest
+	err = json.Unmarshal(data, &manifest)
+	require.NoError(t, err)
+
+	return manifest.BackupMethod
+}
+
+func getLastBackup(t *testing.T) string {
+	output, err := localCluster.VtctldClientProcess.ExecuteCommandWithOutput("GetBackups", shardKsName)
+	require.NoError(t, err)
+
+	// split the backups response into a slice of strings
+	backups := strings.Split(strings.TrimSpace(output), "\n")
+
+	return backups[len(backups)-1]
+}
+
+func TestRestoreIgnoreBackups(t *testing.T) {
+	defer setDefaultCompressionFlag()
+	defer setDefaultCommonArgs()
+	defer cluster.PanicHandler(t)
+
+	cDetails := &CompressionDetails{CompressorEngineName: "pgzip"}
+
+	// launch the custer with xtrabackup as the default engine
+	code, err := LaunchCluster(XtraBackup, "xbstream", 0, cDetails)
+	require.Nilf(t, err, "setup failed with status code %d", code)
+
+	defer TearDownCluster()
+
+	localCluster.DisableVTOrcRecoveries(t)
+	defer func() {
+		localCluster.EnableVTOrcRecoveries(t)
+	}()
+	verifyInitialReplication(t)
+
+	// lets take two backups, each using a different backup engine
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", "--backup-engine=builtin", primary.Alias)
+	require.NoError(t, err)
+	// firstBackup := getLastBackup(t)
+
+	err = localCluster.VtctldClientProcess.ExecuteCommand("Backup", "--allow-primary", "--backup-engine=xtrabackup", primary.Alias)
+	require.NoError(t, err)
+
+	//  insert more data on the primary
+	_, err = primary.VttabletProcess.QueryTablet("insert into vt_insert_test (msg) values ('test2')", keyspaceName, true)
```

Comment:
Ideally insert data that is meaningful to the test. So instead of `"test2"` insert a `"right after xtrabackup backup"` (or whatever makes sense here).

Reply:
I don't see `test2` being tested. I assume that if we insert this value after the full backup, we want to validate that it does not exists after restore.

Reply:
mostly because there are other cases where we validate on vitess by checking numbers of rows, but I agree it would be better testing the values themselves, so I added that